src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -899,29 +900,31 @@ internal void ResolveConflict([NotNull]string docID, [CanBeNull]IConflictResolve
                             remoteDoc.RevID);
 
                     conflictResolver = conflictResolver ?? new DefaultConflictResolver();
-                    var conflict = new Conflict(localDoc.IsDeleted ? null : localDoc, remoteDoc.IsDeleted ? null : remoteDoc);
-                    using (var resolvedDoc = conflictResolver.Resolve(conflict)) {
-                        if (resolvedDoc != null) {
-                            if (resolvedDoc.Id != docID)
-                                throw new InvalidOperationException($"Resolved docID {resolvedDoc.Id} does not match docID {docID}");
-                            else if (resolvedDoc.Database == null)
-                                resolvedDoc.Database = this;
-                            else if (resolvedDoc.Database != this)
-                                throw new InvalidOperationException($"Resolved document db {resolvedDoc.Database.Name} is different from expected db {this.Name}");
-                        }
+                    var conflict = new Conflict(docID, localDoc.IsDeleted ? null : localDoc, remoteDoc.IsDeleted ? null : remoteDoc);
 
-                        InBatch(() =>
-                        {
-                            success = SaveResolvedDocument(resolvedDoc, localDoc, remoteDoc);
-                        });
+                    resolvedDoc = conflictResolver.Resolve(conflict)?.ToMutable();
```

Comment:
This means that the original `resolvedDoc` will be undisposed.  It's not a bug, but better to try to dispose it.  Something like this will be better:

```c#
resolvedDoc = conflictResolver.Resolve(conflict);
Misc.SafeSwap(ref resolvedDoc, resolvedDoc?.ToMutable());
```